src/native/managed/cdacreader/src/Data/TypeHandleArray.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Diagnostics.DataContractReader.Contracts;
+
+namespace Microsoft.Diagnostics.DataContractReader.Data;
+
+internal sealed class TypeHandleArray : IData<TypeHandleArray, (TargetPointer ptr, int size)>
```

Comment:
Same here. It feels like this belongs together with the RuntimeTypeSystem contract.

Maybe we need for the Target to provide some kind of caching service for contracts' auxiliary data?  But it seems like it should be something disctinct from the mechanism that reads/caches target process memory.

Reply:
It all feels like `ProcessedData` to me. We could have 2 different objects on the target, but I don't think that buys us anything. What I think would be useful is to move these sorts of Data things from being held in the Data directory, to being contained within the file that implements the contract.